### PR TITLE
docs(dom): Document breaking changes of v8

### PR DESCRIPTION
**Pull request by eps1lon**
Documentation for https://github.com/testing-library/dom-testing-library/pull/979

### CODE REVIEW

File: docs/dom-testing-library/api-async.mdx [hunk 1]
```diff
@@ -55,7 +55,8 @@ await waitFor(() => expect(mockAPI).toHaveBeenCalledTimes(1))
 // ...
 ```
 
-`waitFor` may run the callback a variable number of times.
+`waitFor` may run the callback a number of times until the timeout is reached.
+Note that the number of calls is constraint by the `timeout` and `interval` options.
```
Comment (kentcdodds) [inline docs/dom-testing-library/api-async.mdx:59]:
```suggestion
Note that the number of calls is constrained by the `timeout` and `interval` options.
```

**Review by kentcdodds — APPROVED** [review]
Other than one typo everything's great 👍

File: docs/dom-testing-library/api-debugging.mdx [hunk 2]
```diff
@@ -51,6 +51,7 @@ It is defined as:
 
 ```typescript
 function prettyDOM(
+  filterNode?: (node: Node) => boolean,
```
Comment (timdeschryver) [inline docs/dom-testing-library/api-debugging.mdx:54]: I believe that this is wrong.
By taking a look at the [type definition](https://github.com/eps1lon/dom-testing-library/blob/alpha/types/pretty-dom.d.ts), I think that `filterNode` is a property of `options`?

Comment (eps1lon) [reply docs/dom-testing-library/api-debugging.mdx]: Thanks! Hope this is clearer now: [`df64629` (#866)](https://github.com/testing-library/testing-library-docs/pull/866/commits/df64629dbd4ee518491439b4caa13c0e11446998)

Comment (timdeschryver) [reply docs/dom-testing-library/api-debugging.mdx]: For me it is, thanks!

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**